plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/AltibaseMetaModel.java

```diff
@@ -97,9 +98,19 @@ public GenericTableBase createTableImpl(@NotNull JDBCSession session, @NotNull G
             case "VIEW":
             case "SYSTEM VIEW":
                 table = new AltibaseView(owner, tableName, tableType, dbResult);
+                try {
+                    ((AltibaseView) table).refreshState(session);
```

Comment:
Do you really need to refresh its state during the creation? The rule of thumb is that you don't want to query anything from the database if your object is not persisted.

Reply:
Actually, I don't remember much about this because I fixed it early this year. I'll look back on my memory and fix it :)

Reply:
@ ShadelessFox,

As you know that because AltibaseMetaModel inherits GenericMetaModel, it retrieves information to create a View or MView using the DatabaseMetaData.getTable method. Since this method does not provide View or MView state information, a separate method was needed, so the refreshState method was provided.

Do you think it's better to handle it inside the View or MView constructor?

Reply:
Thanks for the clarification. I'd put this logic inside constructors.

Reply:
Okay, I modified the code as your recommendation. Looks better :)